rd_ui/app/scripts/visualizations/chart.js

```diff
@@ -27,11 +27,41 @@
         queryResult: '=',
         options: '=?'
       },
-      template: "<chart options='chartOptions' series='chartSeries' class='graph'></chart>",
+      templateUrl: '/views/visualizations/chart.html',
       replace: false,
       controller: ['$scope', function ($scope) {
         $scope.chartSeries = [];
         $scope.chartOptions = {};
+        $scope.dateRangeEnabled = $scope.options.xAxis && $scope.options.xAxis.type === 'datetime';
+        $scope.dateRange = { min: moment('1970-01-01'), max: moment() };
+
+        /**
+         * Update date range by finding date extremes
+         *
+         * ISSUE: chart.getExtreme() does not support getting Moment object out of box
+         * TODO: Find a faster way to do this
+         */
+        var setDateRangeToExtreme = function (allSeries) {
+          if ($scope.dateRangeEnabled && allSeries && allSeries.length > 0) {
+            var maxDateRange = moment('1970-01-01'),
+              minDateRange = moment();
+            _.each(allSeries, function (s) {
+              _.each(s.data, function (point) {
+                if (point.x.isBefore(minDateRange)) {
+                  // Use the copy of point.x to prevent side effects
+                  minDateRange = moment(point.x);
+                }
+                if (point.x.isAfter(maxDateRange)) {
+                  maxDateRange = moment(point.x);
+                }
+              });
+            });
```

Comment:
This logic can be simplified by using [`_.min`](http://underscorejs.org/#min), [`_.max`](http://underscorejs.org/#max) and [`_.pluck`](http://underscorejs.org/#pluck).